controllers/managedocs_controller_test.go

```diff
@@ -278,6 +291,28 @@ var _ = Describe("ManagedOCS controller", func() {
 			}
 		}
 
+		// Race condition:
+		// Begin with the prometheus uninstall condition as false, and all others are true.
+		// Then change to the alertmanager uninstall condition to false; the rest are true.
+		// The following has occurred, causing uninstall tests that shouldn't delete managedocs to delete them:
+		//
+		// UninstallConditions                      Reconciler
+		//------------------------------------------------------------
+		//
+		// StorageCluster Updated
+		// ------------------------------->         Reconcile Loop triggered
+		// Prometheus state set to Ready
+		// ------------------------------->         updateComponentStatus() called, status of all components are Ready
+		// alertmanager state set to Pending
+		// deletion ConfigMap created
+		// ------------------------------->         checkUninstallCondition() called, passes
+		// ------------------------------->         areComponentsReadyForUninstall() called, passes.
+		// ------------------------------->         managedOCS resource is deleted.
+		//
+		// Adding this sleep before creating the configmap mitigates this from happening by preventing
+		// all of the changes from occuring in the middle of a single reconcile loop.
+		time.Sleep(time.Millisecond * 25)
+
```

Comment:
The entire setup process (func) is protected by lines 212-214:

```
		// Delete the configmap to ensure that we will not trigger uninstall accidentally
		// via and intermediate state
		configMap := addonConfigMapTemplate.DeepCopy()
		err := k8sClient.Delete(ctx, configMap)
		Expect(err == nil || errors.IsNotFound(err)).Should(BeTrue())
```

This ensures that for the entire duration of the func the config map does not exist which means uninstall will fail on this condition. 
		
Any of the changes in this function cannot be observed before the recreation of the config map (last step in the process) which means that the only reconcile iteration that matters is the one that observed the existence of the configmap. As components status is always updated at the start of the reconcile loop it means that the situation described here cannot happen.

It might be that I am missing something here, if so please reply with an explanation. But in any case, an arbitrary sleep is not a solution as we can never know how long after k8s DB is updated, a watch event will be triggered. And as a general note, a watch trigger timing has no guarantees or bounds, the only guarantee is that it will eventually fire.

Reply:
> This ensures that for the entire duration of the func the config map does not exist which means uninstall will fail on this condition.

The order of events listed in the comment describes a scenario where this is not the case. This is what was observed to have happened.